### ISSUE TITLE
BakedLightmap crash on large scenes

### ISSUE THREAD

**Issue by clayjohn** [godotengine/godot#38685]
**Godot version:**
Master 66b0b0c

**OS/device including version:**
Windows 10 NVidia GTX 1050 2gb

**Issue description:**
When baking lightmaps in Calinou's sponza scene I get a GPU crash
```
The instruction at 0x00007FF8A3745D6B referenced memory at 0x0000000000000068. The memory could not be read.
```
Along with the following backtrace:
```
CrashHandlerException: Program crashed
Dumping the backtrace. Please include this when reporting the bug on https://github.com/godotengine/godot/issues
[0] vkGetInstanceProcAddr
[1] vkCmdCopyImageToBuffer (C:\Users\Clay\dev\godot\thirdparty\vulkan\loader\trampoline.c:1785)
[2] RenderingDeviceVulkan::texture_get_data (C:\Users\Clay\dev\godot\drivers\vulkan\rendering_device_vulkan.cpp:2511)
[3] RasterizerSceneRD::bake_render_uv2 (C:\Users\Clay\dev\godot\servers\rendering\rasterizer_rd\rasterizer_scene_rd.cpp:4237)
[4] RenderingServerScene::bake_render_uv2 (C:\Users\Clay\dev\godot\servers\rendering\rendering_server_scene.cpp:3018)
[5] RenderingServerRaster::bake_render_uv2 (C:\Users\Clay\dev\godot\servers\rendering\rendering_server_raster.h:637)
[6] RenderingServerWrapMT::bake_render_uv2 (C:\Users\Clay\dev\godot\servers\rendering\rendering_server_wrap_mt.h:543)
[7] BakedLightmap::bake (C:\Users\Clay\dev\godot\scene\3d\baked_lightmap.cpp:699)
[8] BakedLightmapEditorPlugin::_bake_select_file (C:\Users\Clay\dev\godot\editor\plugins\baked_lightmap_editor_plugin.cpp:40)
[9] call_with_variant_args_helper<BakedLightmapEditorPlugin,String const &,0> (C:\Users\Clay\dev\godot\core\callable_method_pointer.h:136)
[10] call_with_variant_args<BakedLightmapEditorPlugin,String const &> (C:\Users\Clay\dev\godot\core\callable_method_pointer.h:162)
[11] CallableCustomMethodPointer<BakedLightmapEditorPlugin,String const &>::call (C:\Users\Clay\dev\godot\core\callable_method_pointer.h:178)
[12] Callable::call (C:\Users\Clay\dev\godot\core\callable.cpp:50)
[13] Object::emit_signal (C:\Users\Clay\dev\godot\core\object.cpp:1201)
[14] Object::emit_signal (C:\Users\Clay\dev\godot\core\object.cpp:1258)
[15] EditorFileDialog::_action_pressed (C:\Users\Clay\dev\godot\editor\editor_file_dialog.cpp:463)
[16] call_with_variant_args_helper<EditorFileDialog> (C:\Users\Clay\dev\godot\core\callable_method_pointer.h:140)
[17] call_with_variant_args<EditorFileDialog> (C:\Users\Clay\dev\godot\core\callable_method_pointer.h:162)
[18] CallableCustomMethodPointer<EditorFileDialog>::call (C:\Users\Clay\dev\godot\core\callable_method_pointer.h:178)
[19] Callable::call (C:\Users\Clay\dev\godot\core\callable.cpp:50)
[20] Object::emit_signal (C:\Users\Clay\dev\godot\core\object.cpp:1201)
[21] Object::emit_signal (C:\Users\Clay\dev\godot\core\object.cpp:1258)
[22] AcceptDialog::_ok_pressed (C:\Users\Clay\dev\godot\scene\gui\dialogs.cpp:110)
[23] call_with_variant_args_helper<AcceptDialog> (C:\Users\Clay\dev\godot\core\callable_method_pointer.h:140)
[24] call_with_variant_args<AcceptDialog> (C:\Users\Clay\dev\godot\core\callable_method_pointer.h:162)
[25] CallableCustomMethodPointer<AcceptDialog>::call (C:\Users\Clay\dev\godot\core\callable_method_pointer.h:178)
[26] Callable::call (C:\Users\Clay\dev\godot\core\callable.cpp:50)
[27] Object::emit_signal (C:\Users\Clay\dev\godot\core\object.cpp:1201)
[28] Object::emit_signal (C:\Users\Clay\dev\godot\core\object.cpp:1258)
[29] BaseButton::_pressed (C:\Users\Clay\dev\godot\scene\gui\base_button.cpp:135)
[30] BaseButton::on_action_event (C:\Users\Clay\dev\godot\scene\gui\base_button.cpp:174)
[31] BaseButton::_gui_input (C:\Users\Clay\dev\godot\scene\gui\base_button.cpp:64)
[32] MethodBind1<BaseButton,Ref<InputEvent> >::call (C:\Users\Clay\dev\godot\core\method_bind.gen.inc:867)
[33] Object::call_multilevel (C:\Users\Clay\dev\godot\core\object.cpp:747)
[34] Object::call_multilevel (C:\Users\Clay\dev\godot\core\object.cpp:848)
[35] Viewport::_gui_call_input (C:\Users\Clay\dev\godot\scene\main\viewport.cpp:1675)
[36] Viewport::_gui_input_event (C:\Users\Clay\dev\godot\scene\main\viewport.cpp:1997)
[37] Viewport::input (C:\Users\Clay\dev\godot\scene\main\viewport.cpp:3035)
[38] Window::_window_input (C:\Users\Clay\dev\godot\scene\main\window.cpp:904)
[39] call_with_variant_args_helper<Window,Ref<InputEvent> const &,0> (C:\Users\Clay\dev\godot\core\callable_method_pointer.h:136)
[40] call_with_variant_args<Window,Ref<InputEvent> const &> (C:\Users\Clay\dev\godot\core\callable_method_pointer.h:162)
[41] CallableCustomMethodPointer<Window,Ref<InputEvent> const &>::call (C:\Users\Clay\dev\godot\core\callable_method_pointer.h:178)
[42] Callable::call (C:\Users\Clay\dev\godot\core\callable.cpp:50)
[43] DisplayServerWindows::_dispatch_input_event (C:\Users\Clay\dev\godot\platform\windows\display_server_windows.cpp:1779)
[44] DisplayServerWindows::_dispatch_input_events (C:\Users\Clay\dev\godot\platform\windows\display_server_windows.cpp:1762)
[45] Input::_parse_input_event_impl (C:\Users\Clay\dev\godot\core\input\input.cpp:561)
[46] Input::parse_input_event (C:\Users\Clay\dev\godot\core\input\input.cpp:379)
[47] Input::flush_accumulated_events (C:\Users\Clay\dev\godot\core\input\input.cpp:793)
[48] DisplayServerWindows::process_events (C:\Users\Clay\dev\godot\platform\windows\display_server_windows.cpp:1520)
[49] OS_Windows::run (C:\Users\Clay\dev\godot\platform\windows\os_windows.cpp:661)
[50] widechar_main (C:\Users\Clay\dev\godot\platform\windows\godot_windows.cpp:162)
[51] _main (C:\Users\Clay\dev\godot\platform\windows\godot_windows.cpp:184)
[52] main (C:\Users\Clay\dev\godot\platform\windows\godot_windows.cpp:196)
[53] __scrt_common_main_seh (d:\agent\_work\3\s\src\vctools\crt\vcstartup\src\startup\exe_common.inl:288)
[54] BaseThreadInitThunk
-- END OF BACKTRACE --
```

**Steps to reproduce:**
1. Download https://github.com/Calinou/godot-sponza-4
2. Unwrap UVs for lightmap
3. Add a BakedLightmap to the scene
4. Press "Bake"

**Minimal reproduction project:**
I can't reproduce on a small scene, but I can reproduce in https://github.com/Calinou/godot-sponza-4

**Comment by clayjohn** [comment]
Running it on small scenes with 1-2 small meshes seems to work okay. Running it on GLTF sponza (only 50mb vs Calinou's sponza which is 400mb) results in this modern art piece:

![Screenshot (174)](https://user-images.githubusercontent.com/16521339/81646918-f3c5b800-93e0-11ea-8862-c242bec90497.png)

Clearly we have some issue managing our memory in VRAM.

**Comment by Calinou** [comment]
My new Sponza demo is broken, it should have been made as an inherited scene to be fair.

**Comment by clayjohn** [comment]
@Calinou I don't think the issue is with your scene. I think we are somehow receiving an invalid memory pointer when allocating buffers for baking the lightmaps. Likely some combination of large scene and low amount of VRAM.

**Comment by akien-mga** [comment]
Fixed by #44628.
